File: results/java/SMALLEST_ELEMENT_REPEATED_EXACTLY_K_TIMES_NOT_LIMITED_SMALL_RANGE.cpp
int f_filled ( int a [ ] , int n , int k ) {
  unordered_map < int , int > m ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) if ( m . count ( a [ i ] ) ) m [ a [ i ] ] = m [ a [ i ] ] + 1 ;
  else m [ a [ i ] ] = 1 ;
  int res = INT_MAX ;
  unordered_set < int > :: iterator it = m . begin ( ) ;
  for ( int temp : it ) if ( m [ temp ] == k ) res = min ( res , temp ) ;
  return ( res != INT_MAX ) ? res : - 1 ;
}

File: results/java_codex/SMALLEST_ELEMENT_REPEATED_EXACTLY_K_TIMES_NOT_LIMITED_SMALL_RANGE.java
int smallestKFreq(int a[], int n, int k) {
        unordered_map<int, int> m;
        for (int i = 0; i < n; i++)
            if (m.find(a[i]) != m.end())
                m[a[i]]++;
            else
                m[a[i]] = 1;
        int res = INT_MAX;
        for (auto it = m.begin(); it != m.end(); it++)
            if (it->second == k)
                res = min(res, it->first);
        return (res != INT_MAX) ? res : -1;
    }